plugins/inputs/f5_load_balancer/f5_load_balancer.go

```diff
@@ -0,0 +1,343 @@
+			fields["net_interface_counter_packets_in"], _ = strconv.Atoi(gjson.Get(base, "counters\\.pktsIn.value").Raw)
+			fields["net_interface_counter_packets_out"], _ = strconv.Atoi(gjson.Get(base, "counters\\.pktsOut.value").Raw)
+			status := 0
+			if gjson.Get(base, "status.description").Str == "up" {
+				status = 1
+			}
+			fields["net_interface_status"] = status
+
+			acc.AddFields("f5_load_balancer", fields, tags)
+		}(&wg, url, acc)
+	}
+	wg.Wait()
+}
+
+func (f5 *F5LoadBalancer) GetTags(endpoint string) (string, map[string]string, error) {
+	resp, err := f5.Call(endpoint)
+	tags := map[string]string{}
+	if err != nil {
+		return resp, tags, err
+	}
+	selfLinkSplit := strings.Split(gjson.Get(resp, "selfLink").Str, "~")
+	if len(selfLinkSplit) > 2 {
+		selfLinkSplit = strings.Split(selfLinkSplit[2], "/")
+		if len(selfLinkSplit) > 0 {
+			tags["name"] = selfLinkSplit[0]
+		}
+	}
+	if _, ok := tags["name"]; !ok {
+		return resp, tags, fmt.Errorf("Bad or malformed response")
+	}
+	return resp, tags, nil
+}
+
+func (f5 *F5LoadBalancer) GetUrls(endpoint string) ([]string, error) {
+	resp, err := f5.Call(endpoint)
+	urls := make([]string, 0, 2)
+	if err != nil {
+		return urls, err
+	}
+	for _, item := range gjson.Get(resp, "items").Array() {
+		selfLink := gjson.Get(item.Raw, "selfLink").Str
+		urls = append(urls, strings.Split(strings.Split(selfLink, "localhost")[1], "?")[0]+"/stats")
+	}
+	return urls, nil
+}
+
+func (f5 *F5LoadBalancer) Call(endpoint string) (string, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", f5.URL+endpoint, nil)
+	if err != nil {
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-F5-Auth-Token", f5.Token)
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+func (f5 *F5LoadBalancer) Authenticate() error {
+	f5.Token = ""
+	body := map[string]string{"username": f5.Username, "password": f5.Password, "loginProviderName": "tmos"}
+	jsonBody, _ := json.Marshal(body)
+	client := &http.Client{}
+	req, err := http.NewRequest("POST", f5.URL+"/mgmt/shared/authn/login", bytes.NewBuffer(jsonBody))
+	if err != nil {
+		return err
+	}
+	req.SetBasicAuth(f5.Username, f5.Password)
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	jsonString := string(data)
+	f5.Token = gjson.Get(jsonString, "token.token").Str
+	return nil
+}
+
+func (f5 *F5LoadBalancer) ResetToken() {
+	f5.Token = ""
+}
+
+func contains(s []string, str string) bool {
+	for _, v := range s {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
+func init() {
+	inputs.Add("f5_load_balancer", func() telegraf.Input {
+		return &F5LoadBalancer{}
```

Comment:
Initializing the collectors default values here would be better, then you won't have to check for nil in the struct init method.